core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalysis.java

```diff
@@ -50,7 +52,8 @@ public ExpressionAnalysis(
             Map<NodeRef<Expression>, ResolvedField> columnReferences,
             Set<NodeRef<Expression>> typeOnlyCoercions,
             Set<NodeRef<QuantifiedComparisonExpression>> quantifiedComparisons,
-            Set<NodeRef<FunctionCall>> windowFunctions)
+            Set<NodeRef<FunctionCall>> windowFunctions,
+            Map<NodeRef<Expression>, ResolvedFunction> resolvedFunctions)
```

Comment:
@dain @martint please take a look

Reply:
@martint 
- As your mentioned "for instance, lack of support for constant expressions that contain lambda expressions.", could you help to give me a example SQL to reproduce it. so I will try to fix it asap.
- For IR expression. i still need more time to learn.